### PR TITLE
Extract code from views by creating an ActionButton component

### PR DESCRIPTION
**Pull request by jcoyne**
## Why was this change made?

Keep code out of the views.

## Was the documentation updated?

N/A

### CODE REVIEW

**Comment by jcoyne** [comment]
@mjgiarlo as you can see by the test failure, this component is adequately covered by the view spec.  If you want to add coverage, adding a component spec is a good way.

**Comment by mjgiarlo** [comment]
@jcoyne :speech_balloon: 
> @mjgiarlo as you can see by the test failure, this component is adequately covered by the view spec. If you want to add coverage, adding a component spec is a good way.

I don't so much care about the overall coverage. I care more about getting a sense of whether folks would rather sling view specs or component specs (or both) going forward, as we convert views to being backed by components. Maybe that's not the right question to ask, though.

**Comment by peetucket** [comment]
I don't think both kinds of tests are really providing a benefit.  Having to pick one, I'd do a component spec, but don't think I would both converting when a refactoring like this is done.

File: app/components/action_button.rb [hunk 1]
```diff
@@ -0,0 +1,32 @@
+# frozen_string_literal: true
+
+# Draws a blue button in the side menu
+class ActionButton < ApplicationComponent
+  # @param [Hash] properties the button properties
+  def initialize(label:, url:, confirm: nil, new_page: nil, check_url: nil, disabled: nil)
+    @label = label
```
Comment (peetucket) [inline app/components/action_button.rb]: it feels like there ought to be a more elegant way to avoid so much boilerplate code like this

Comment (jcoyne) [reply app/components/action_button.rb]: We could make a struct, but that's just moving the boilerplate, isn't it?

**Pull request merged**